lib/matplotlib/testing/decorators.py

```diff
@@ -483,7 +508,36 @@ def _image_directories(func):
     doesn't exist.
     """
     module_path = Path(sys.modules[func.__module__].__file__)
-    baseline_dir = module_path.parent / "baseline_images" / module_path.stem
+    cache_dir = Path(get_cache_dir()) / "baseline_images" / module_path.stem
+    if func.__module__.startswith("matplotlib."):
+        try:
+            import matplotlib_baseline_images
+            baseline_dir = (Path(matplotlib_baseline_images.__file__).parent /
+                            module_path.stem)
+        except ImportError:
+            baseline_dir = (module_path.parent /
+                            "baseline_images" /
+                            module_path.stem)
+
+    elif func.__module__.startswith("mpl_toolkits."):
+        try:
+            import mpl_toolkits_baseline_images
+            baseline_file = mpl_toolkits_baseline_images.__file__
+            baseline_dir = (Path(baseline_file).parent /
+                            module_path.stem)
+        except ImportError:
+            baseline_dir = (module_path.parent /
+                            "baseline_images" /
+                            module_path.stem)
+    for baseline_img in Path(baseline_dir).glob('**/*.py'):
+        baseline_dir_img = baseline_dir / baseline_img
+        cache_dir_img = cache_dir / baseline_img
+        if not cache_dir_img.exists():
+            cache_dir_img.mkdir(parents=True, exist_ok=True)
```

Comment:
Can probably move this outside of the loop?

Reply:
This is on the per image basis, so it has to be inside the loop I guess.